types/global.d.ts

```diff
@@ -9,6 +9,43 @@ declare class Platform {
   closeCurrentWindow: () => void;
 }
 
+declare class MessageSender {
+  documentId?: string;
+
+  documentLivecycle?: string;
+
+  frameId?: number;
+
+  id?: string;
+
+  origin?: string;
+
+  url?: string;
+}
+
+declare class Runtime {
+  onMessage: {
+    addListener: (
+      callback: (
+        message: any,
+        sender: MessageSender,
+        sendResponse: (response: any) => void,
+      ) => void,
+    ) => void;
+  };
+
+  sendMessage: (
+    extensionId?: string,
+    message: any,
+    options?: Record<string, unknown>,
+    callback?: (response: any) => void,
+  ) => void;
+}
+
+declare class Chrome {
```

Comment:
non-blocking comment:

If we don't want to maintain these types ourselves `@types/chrome` contains the `Runtime` type, which I'm also planning on using in the build system port to webpack/esbuild. To use it:

`yarn add @types/chrome --dev`

and then in the files that need it add

```
/// <reference path="../node_modules/@types/chrome/index.d.ts"/>
```

or you can add it to our tsconfig's `compilerOptions.types` array: https://www.typescriptlang.org/tsconfig#types. This is convenient, but the downside is that these types then become available in all JS contexts, which I imaging don't all have these `chrome` globals actually available.

Reply:
@davidmurdoch i'll probably do something with this in the trezor PR